Shoko.Server/API/v3/Models/Shoko/Renamer.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Newtonsoft.Json;
+using Shoko.Models.Server;
+using Shoko.Server.API.v3.Models.Common;
+using Shoko.Server.Repositories;
+using Shoko.Server.Settings;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Linq;
+
+namespace Shoko.Server.API.v3.Models.Shoko
+{
+    public class Renamer : BaseModel
+    {
+        public Renamer(RenameFileHelper.RenamerInfo renamerInfo)
+        {
+            var scripts = RepoFactory.RenameScript.GetRenamerType(renamerInfo.Name);
+            ID = renamerInfo.ID;
+            Name = renamerInfo.Name;
+            Size = scripts.Count;
+            Description = renamerInfo.Description;
+            Priority = ServerSettings.Instance.Plugins.Priority.Contains(Name) ? ServerSettings.Instance.Plugins.Priority.IndexOf(Name) : null;
+            Enabled = ServerSettings.Instance.Plugins.EnabledRenamers.ContainsKey(Name) ? ServerSettings.Instance.Plugins.EnabledRenamers[Name] : true;
+        }
+
+        /// <summary>
+        /// Renamer ID.
+        /// </summary>
+        public int ID { get; set; }
+
+
+        /// <summary>
+        /// A short description about the renamer.
+        /// </summary>
+        public string Description { get; set; }
+
+
+        /// <summary>
+        /// If the renamer is enabled.
+        /// </summary>
+        public bool Enabled { get; set; }
+
+        /// <summary>
+        /// Lower numbers mean higher priority. Will be null if a priority is not set yet.
+        /// </summary>
+        public int? Priority { get; set; }
+
+        public class Script
+        {
+            public Script()
+            {
+            }
+
+            public Script(RenameScript script)
+            {
+                RenameFileHelper.RenamerInfo renamerInfo;
+                RenameFileHelper.Renamers.TryGetValue(script.RenamerType, out renamerInfo);
+                IDs = new RenamerScriptIDs() { ID = script.RenameScriptID, Renamer = renamerInfo?.ID };
+                RenamerName  =  script.RenamerType;
+                Name = script.ScriptName;
+                EnabledOnImport = script.IsEnabledOnImport == 1;
+                Body = script.Script;
+            }
+
+            public void MergeWithExisting(RenameScript existing)
+            {
+                if (IDs.Renamer.HasValue)
+                {
+                    var renamerInfo = RenameFileHelper.Renamers.Values.FirstOrDefault(r => r.ID == IDs.Renamer.Value);
+                    if (renamerInfo != null)
+                        existing.RenamerType = renamerInfo.Name;
```

Comment:
Why pull it from an arbitrary name when the string type is what was already used?